packages/plugins/src/cli/index.ts

```diff
@@ -6,4 +6,38 @@ import { makeCLI } from './setup'
 // Load environment variables from .env
 dotenv.config()
 
+let getaddrinfoErrors = 0
+// If an error occurs, we write the error message and stack trace to `stdout` then exit the process
+// with exit code `1`. We write the error to `stdout` instead of `stderr` because `stderr` may
+// contain warnings that were written via `console.warn`, which are indistinguishable from the
+// actual error message in `stderr`. By using `stdout`, we can throw an error that doesn't contain
+// warnings in the parent process.
+process.on('uncaughtException', (error) => {
```

Comment:
We still need to catch `getaddrinfo ENOTFOUND` errors, but since we no longer invoke the hardhat runner as a separate script, we have to set this up when the CLI is first called.